### PR TITLE
Update the CRD for kopia integration

### PR DESCRIPTION
**Pull request by reasonerjt**
This commit adds additional fields to podvolumebackup and podvolumerestore.  The resticrepository will be renamed to
backuprepository

Signed-off-by: Daniel Jiang <jiangd@vmware.com>

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [X] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [X] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Comment by sseago** [comment]
We'll need to be careful with the resticrepository->backuprepository change on upgrade. Do we need to delete the old CRD on upgrade? Any problems with existing installs being upgraded in terms of what was in the prior resticrepository metadata, or will everything just get recreated from scratch as necessary?

**Comment by reasonerjt** [comment]
@sseago it has been discussed and covered in the design
https://github.com/vmware-tanzu/velero/pull/4926
in particular this section:
https://github.com/vmware-tanzu/velero/pull/4926/files#diff-d96d1929fa6c873a7fd9174256bdfecdfbb5acdf28a398d7905c42720bfc6224R311

**Comment by sseago** [comment]
@reasonerjt Ahh thanks. I remember reading that now when I went over this the first time, but I'd forgotten. So it's covered.

File: pkg/apis/velero/v1/pod_volume_backup_types.go [hunk 1]
```diff
@@ -34,12 +34,17 @@ type PodVolumeBackupSpec struct {
...
 	BackupStorageLocation string `json:"backupStorageLocation"`
 
-	// RepoIdentifier is the restic repository identifier.
+	// RepoIdentifier is the backup repository identifier.
 	RepoIdentifier string `json:"repoIdentifier"`
 
+	// UploaderType is the type of the uploader to handle the data transfer.
+	// +kubebuilder:validation:Enum=kopia;restic;""
+	// +optional
+	UploaderType string `json:"uploader-type"`
```
Comment (ywk253100) [inline pkg/apis/velero/v1/pod_volume_backup_types.go:46]: Seems the convention is to use `camelCase` in the `json` tag

File: pkg/apis/velero/v1/pod_volume_restore_type.go [hunk 2]
```diff
@@ -30,12 +30,17 @@ type PodVolumeRestoreSpec struct {
...
 	BackupStorageLocation string `json:"backupStorageLocation"`
 
-	// RepoIdentifier is the restic repository identifier.
+	// RepoIdentifier is the backup repository identifier.
 	RepoIdentifier string `json:"repoIdentifier"`
 
+	// UploaderType is the type of the uploader to handle the data transfer.
+	// +kubebuilder:validation:Enum=kopia;restic;""
+	// +optional
+	UploaderType string `json:"uploader-type"`
```
Comment (ywk253100) [inline pkg/apis/velero/v1/pod_volume_restore_type.go:42]: Seems the convention is to use `camelCase` in the `json` tag

Comment (reasonerjt) [reply pkg/apis/velero/v1/pod_volume_restore_type.go:42]: I don't disagree, but it is done following the design.  
@Lyndon-Li I also think we should use camelCase here, if it sounds good to you, I'll make the change and also update the design in this PR.

Let me know your thoughts

Comment (Lyndon-Li) [reply pkg/apis/velero/v1/pod_volume_restore_type.go:42]: @reasonerjt It sounds good to me. Please help to update the design. Thanks.

File: config/crd/v1/bases/velero.io_backuprepositories.yaml [hunk 3]
```diff
@@ -48,11 +48,11 @@ spec:
                   be run.
                 type: string
               resticIdentifier:
-                description: ResticIdentifier is the full restic-compatible string
+                description: ResticIdentifier is the full backup-compatible string
                   for identifying this repository.
                 type: string
               volumeNamespace:
```
Comment (Lyndon-Li) [inline config/crd/v1/bases/velero.io_backuprepositories.yaml:54]: Looks like the repositoryType is missing in this CRD?

File: pkg/restic/backupper.go [hunk 4]
```diff
@@ -274,6 +274,7 @@ func newPodVolumeBackup(backup *velerov1api.Backup, pod *corev1api.Pod, volume c
 			},
 			BackupStorageLocation: backup.Spec.StorageLocation,
 			RepoIdentifier:        repoIdentifier,
+			UploaderType:          "restic",
```
[the remark this answers is not in the document]
Comment (reasonerjt) [reply pkg/restic/backupper.go]: correct, I forgot to make the update when I decided to make it a parameter.

**Review by ywk253100 — APPROVED** [review]

**Review by Lyndon-Li — APPROVED** [review]

**Pull request merged**